### PR TITLE
added ability to edit feature description

### PR DESCRIPTION
**Pull request by federicomaffei**
@christriddle available for code review whenever you're free. I think would be nice to add some e2e tests related to descriptions, as a next step. Did not want to add them now, since it is already a somewhat big PR.

### CODE REVIEW

**Comment by christriddle** [comment]
@federicomaffei I think it would be a good idea to add the e2e tests as part of this PR. That way it's a good unit.

File: server/routes/featureRoutes.js [hunk 1]
```diff
@@ -42,12 +42,25 @@ var addFeature = function(req, res){
...
 
-    feature.updateFeatureToggle(applicationName, featureName, value, req, function(err){
-        if (err) throw err;
-        res.send(200);
-    });
+    if(req.body.newFeatureDescription) {
+        var newFeatureDescription = req.body.newFeatureDescription;
+
+        feature.updateFeatureDescription(applicationName, featureName, newFeatureDescription, req, function(err){
+            console.log('test');
```
Comment (andyroyle) [inline server/routes/featureRoutes.js]: probably shouldn't be here ;)

**Comment by federicomaffei** [comment]
@christriddle np, will add them here.

File: server/routes/featureRoutes.js (hunk 1 above)
Comment (federicomaffei) [inline server/routes/featureRoutes.js]: @andyroyle well spotted, totally missed it.

**Comment by federicomaffei** [comment]
@christriddle added the relevant e2e test, think the PR is ready for review now.

File: client/tests/e2e/update-description.spec.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    afterEach(function(done){
+        removeAllFeatures(done);
+    });
+
+    var getDesc = function(){
+        return element(by.css(descCss));
+    };
+
+    var expectDescToHaveText = function(state){
```
Comment (christriddle) [inline client/tests/e2e/update-description.spec.js]: The state param here isn't doing anything.. I think you meant to pass the expected value in here?

File: server/domain/feature.js [hunk 3]
```diff
@@ -291,6 +302,11 @@ module.exports.updateFeatureToggle = function(applicationName, featureName, valu
     });
 };
 
+module.exports.updateFeatureDescription = function(applicationName, featureName, newFeatureDescription, req, cb){
+    addFeatureDescription(applicationName, featureName, newFeatureDescription);
+    cb();
```
Comment (christriddle) [inline server/domain/feature.js]: This callback should probably be called once the description has been updated in etcd

File: server/routes/featureRoutes.js [hunk 4]
```diff
@@ -42,12 +42,24 @@ var addFeature = function(req, res){
 var updateFeatureToggle = function(req, res){
     var applicationName = req.params.applicationName;
     var featureName = req.params.featureName;
-    var value = req.body.value;
 
-    feature.updateFeatureToggle(applicationName, featureName, value, req, function(err){
-        if (err) throw err;
-        res.send(200);
-    });
+    if(req.body.newFeatureDescription) {
```
Comment (christriddle) [inline server/routes/featureRoutes.js]: So here we are doing two things with the same route, depending on the body? Not sure I like this - suggest we have two different routes maybe?

File: server/domain/feature.js [hunk 5]
```diff
@@ -187,23 +187,34 @@ module.exports.getFeature = function(applicationName, featureName, cb) {
...
-            toggleSuggestions = getToggleSuggestions(metaData, toggles);
-        } else {
-            toggles = getSimpleFeatureToggle(featureName, result.node);
-        }
+        etcd.client.get('v1/metadata/' + applicationName + '/descriptions', function(descriptionError, descriptionResult){
+            if(descriptionError){
+                console.log(descriptionError);
+            }
+
+            var descriptionsMap = !descriptionError ? getDescriptionsMap(descriptionResult.node) : {};
```
Comment (christriddle) [inline server/domain/feature.js]: Maybe we need to move everything below into a separate method, is sort of getting confusing

**Comment by federicomaffei** [comment]
@christriddle addressed all your comments, ready for your review.

File: client/views/partials/application/featureCategory.jade [hunk 6]
```diff
@@ -15,7 +15,8 @@
         tr(ng-repeat="feature in category.features | orderBy: 'name'")
           td
             a(href="/#!/applications/{{applicationName}}/{{feature.name}}") {{feature.name}}
-          td {{feature.description}}
+          td
+            span {{ feature.description || "empty" }}
```
Comment (christriddle) [inline client/views/partials/application/featureCategory.jade]: Maybe the default should just be ""?

File: client/views/partials/feature/toggles.jade [hunk 7]
```diff
@@ -17,6 +18,8 @@
             |  
             a.find-your-code(href="{{toggle.githubSearchUrl}}", target="_blank", tooltip="Find this toogle in your code")
               i.fa.fa-search
+          td
+            span(editable-text="description" onaftersave="updateToggleDescription(featureName, description)") {{ description || "empty" }}
```
Comment (christriddle) [inline client/views/partials/feature/toggles.jade]: Maybe the default should just be ""?

File: server/domain/feature.js [hunk 8]
```diff
@@ -291,6 +316,11 @@ module.exports.updateFeatureToggle = function(applicationName, featureName, valu
     });
 };
 
+module.exports.updateFeature = function(applicationName, featureName, newFeatureDescription, req, cb){
+    addFeatureDescription(applicationName, featureName, newFeatureDescription);
+    cb();
```
Comment (christriddle) [inline server/domain/feature.js]: Still think this empty callback is a bit weird, the cb should be called once the description is changed.

**Comment by christriddle** [comment]
Couple of issues, but other than that, all good in the hood. nice work.

**Comment by federicomaffei** [comment]
@christriddle hopefully done, thanks for the patience dude

**Comment by christriddle** [comment]
@federicomaffei Hey dude, getting an error when when I update a description: 
TypeError: object is not a function
    at /Users/criddle/Code/hobknob/server/domain/feature.js:212:17

Can you see if this works on your machine?

**Comment by federicomaffei** [comment]
@christriddle fixed the bug, ready for review. Cheers

**Pull request merged**